hudi-common/src/main/java/org/apache/hudi/storage/HoodieStorageUtils.java

```diff
@@ -34,22 +32,35 @@ public static HoodieStorage getStorage(StorageConfiguration<?> conf) {
   }
 
   public static HoodieStorage getStorage(FileSystem fs) {
-    return new HoodieHadoopStorage(fs);
+    return (HoodieStorage) ReflectionUtils.loadClass("org.apache.hudi.storage.hadoop.HoodieHadoopStorage", new Class<?>[] {FileSystem.class}, fs);
```

Comment:
Similar here on the class name and reflection utils to construct `HoodieStorage` using reflection.